src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+                    _sb.Append(" | ");
+                    _sb.Append($"REJ60: {(registerValue & 0b0000_0000_0000_0100_0000_0000) >> 10}");
+                    _sb.Append(" | ");
+                    _sb.Append($"FS: {UInt32ToBinaryString((registerValue & 0b0000_0000_0000_0011_1111_1111), 10)}");
+
+                    result = _sb.ToString();
+                    break;
+                case Register.Configuration:
+                    _sb.Clear();
+                    _sb.Append($"Chop: {(registerValue & 0b1000_0000_0000_0000_0000_0000) >> 23}");
+                    _sb.Append(" | ");
+                    _sb.Append($"00: {UInt32ToBinaryString((registerValue & 0b0110_0000_0000_0000_0000_0000) >> 21, 2)}");
+                    _sb.Append(" | ");
+                    _sb.Append($"REFSEL: {(registerValue & 0b0001_0000_0000_0000_0000_0000) >> 20}");
+                    _sb.Append(" | ");
+                    _sb.Append($"0: {(registerValue & 0b0000_1000_0000_0000_0000_0000) >> 19}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Pseudo: {(registerValue & 0b0000_0100_0000_0000_0000_0000) >> 18}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Channel: {UInt32ToBinaryString((registerValue & 0b0000_0011_1111_1111_0000_0000) >> 8, 10)}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Burn: {(registerValue & 0b0000_0000_0000_0000_1000_0000) >> 7}");
+                    _sb.Append(" | ");
+                    _sb.Append($"REFDET: {(registerValue & 0b0000_0000_0000_0000_0100_0000) >> 6}");
+                    _sb.Append(" | ");
+                    _sb.Append($"0: {(registerValue & 0b0000_0000_0000_0000_0010_0000) >> 5}");
+                    _sb.Append(" | ");
+                    _sb.Append($"BUF: {(registerValue & 0b0000_0000_0000_0000_0001_0000) >> 4}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Unipolar: {(registerValue & 0b0000_0000_0000_0000_0000_1000) >> 3}");
+                    _sb.Append(" | ");
+                    _sb.Append($"Gain: {UInt32ToBinaryString((registerValue & 0b0000_0000_0000_0000_0000_0111), 3)}");
+
+                    result = _sb.ToString();
+                    break;
+                default:
+                    result = registerValue.ToString();
+                    break;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Disposes the AD7193 object and closes the SPI device
+        /// </summary>
+        public void Dispose()
+        {
+            if (_spiDevice != null)
+            {
+                _spiDevice?.Dispose();
+                _spiDevice = null;
+            }
+        }
+
+        private uint GetRegisterBits(Register reg, BitMask bitmask)
+        {
+            uint result = _registerCache[(byte)reg] & (uint)bitmask;     // clear all bit(s) except the bit(s) we would like to read
+
+            if (bitmask > 0)
```

Comment:
I'm having hard time following what this piece of code is doing. Seems to me it will remove the last n bits of the result (with n being number of 0s in the binary representation of the bitmask) but what's not clear is why do we do that. Could you write a comment explaining what's happening here?

Reply:
Many device has registers and the registers are compartmentalized into smaller values.
For example in the case of [AD7193](https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf), the 8-bit Status register stores variables like the the 1-bit ready, 1-bit error and 4-bit channel index variables.
To extract these values separately I created the GetRegisterBits method, so I can pass a mask indicating the location of the - let's say - ready bit in the register, and the method returns me only 1-bit of information: 1 if the ready bit is set, 0 is it's not set.
I've just made a description for this method, to make this more understandable.